src/main/java/net/logstash/logback/LogstashFormatter.java

```
@@ -222,6 +216,22 @@ public void setIncludeMdc(boolean includeMdc) {
         }
     }
 
+    public boolean isIncludeKvp() {
+        return this.kvpProvider != null;
+    }
+
+    public void setIncludeKvp(boolean includeKvp) {
+        if (isIncludeKvp() != includeKvp) {
```

let's use fail fast: 
```
if (isIncludeKvp() == includeKvp) {
  return;
}
if (includeKvp) {
   kvpProvider = new KeyValuePairJsonProvider();
     addProvider(kvpProvider);
   } else {
      getProviders().removeProvider(kvpProvider);
       kvpProvider = null;
    }
```